main.go

```diff
@@ -133,6 +134,46 @@ func (d *Downloader) downloadFileWithTimeout(userCtx context.Context, u string)
 	}
 }
 
+func (d *Downloader) getSize(ctx context.Context, u string) (uint64, error) {
+	req, err := http.NewRequestWithContext(ctx, http.MethodHead, u, nil)
+	if err != nil {
+		return 0, fmt.Errorf("Error fetching content size -- creating the request for %s: %w", u, err)
+	}
+	resp, err := d.client.Do(req)
+	if err != nil {
+		return 0, fmt.Errorf("Error fetching content size -- sending get http request to %s: %w", u, err)
+	}
+	if resp.StatusCode != 200 {
+		return 0, fmt.Errorf("error fetching content size -- status code is not valid: %v", resp.StatusCode)
```

Comment:
I know this was @danielfireman's suggestion, but I suggest reverting them.

As I explained [before](https://github.com/cuducos/chunk/pull/10#discussion_r1022900689):

> I think I disagree with this suggested change. The _error fetching content size_ message should be wrapped by the caller of `getSize` in case `getSize` returns an error; not within `getSize`'s implementation (same for the similar suggested change below). Does that make sense?

I mean that let's say `getFileSize` returns an error `error fetching content size  -- status code invalid: 404 Not Found.

Probably I'm gonna wrap the error in getFileSize too to pass to its caller, right? For example:

```go
func download(url string) error { 
    s, err := getFileSize(url)
    if err != nil {
      return fmt.Errorf("error fetching content size: %w", err)
    }
    // etc…
}
```

This would be repetitive for the user as it would become `error fetching content size: error fetching content size  -- status code invalid: 404 Not Found`.

In the context of `getFileSize`, adding the _error fetching content size_ to any error message seems redundant IMHO. This should be something wrapped up by the caller.

```suggestion
		return 0, fmt.Errorf("creating the request for %s: %w", u, err)
	}
	resp, err := d.client.Do(req)
	if err != nil {
		return 0, fmt.Errorf("sending get http request to %s: %w", u, err)
	}
	if resp.StatusCode != 200 {
		return 0, fmt.Errorf("status code for %s not valid: %s", u, resp.Status)
```

Also, I discourage using `%v` when we can either use [an integer (`resp.StatusCode`) or a string (`resp.Status`)](https://cs.opensource.google/go/go/+/refs/tags/go1.19.3:src/net/http/response.go;l=36-37). Finally, I recommend adding the URL to the status code error to facilitate debugging.